packages/engine-ui/src/components/forms/inputs/FleetInput.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react'
+import * as Elements from '../../../shared-elements'
+import * as stores from '../../../utils/state/stores'
+import styled from 'styled-components'
+import arrowIcon from '../../../assets/input-arrowdown.svg'
+
+const ShowDropdownIcon = styled.img`
+  width: 10px;
+  height: 10px;
+`
+const ShowDropdownButton = styled.button`
+  position: absolute;
+  left: 90%;
+  height: 90%;
+  top: 2px;
+  border: none;
+  font-weight: 600;
+  color: ${({ color }) => color || 'inherit'};
+  font-size: 0.875rem;
+  border: none;
+  cursor: pointer;
+  background-color: #f1f3f5;
+  :focus {
+    outline: none;
+  }
+`
+
+const DropdownButtonWithAddButton = styled(Elements.Form.DropdownButton)`
+  display: flex;
+  justify-content: space-between;
+  button {
+    font-weight: 600;
+    color: ${({ color }) => color || 'inherit'};
+    font-size: 0.875rem;
+    border: none;
+    cursor: pointer;
+    background-color: #f1f3f5;
+    :focus {
+      outline: none;
+    }
+  }
+`
+
+const Dropdown: React.FC<{
+  items: string[]
+  searchValue?: string
+  handleDropdownSelect: (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    value: string
+  ) => void
+}> = ({ items, handleDropdownSelect, searchValue }) => {
+  console.log(searchValue)
+  return (
+    <Elements.Form.DropdownWrapper>
+      {searchValue && (
+        <DropdownButtonWithAddButton
+          name={searchValue}
+          onMouseDown={(event) => handleDropdownSelect(event, searchValue)}
+        >
+          {searchValue}
+          <button
+            type="button"
+            onMouseDown={(event) => handleDropdownSelect(event, searchValue)}
+          >
+            +
+          </button>
+        </DropdownButtonWithAddButton>
+      )}
+      {items.map((value: string, index: number) => (
+        <Elements.Form.DropdownButton
+          key={index}
+          name={value}
+          onMouseDown={(event) => handleDropdownSelect(event, value)}
+        >
+          {value}
+        </Elements.Form.DropdownButton>
+      ))}
+    </Elements.Form.DropdownWrapper>
+  )
+}
+
+interface Props {
+  onChangeHandler: any
+  placeholder: string
+  value: string
+}
+
+const Component: React.FC<Props> = ({
+  onChangeHandler,
+  placeholder,
+  value,
+  ...rest
+}) => {
+  const transports = stores.dataState((state) => state.transports)
+  const [fleets, setFleets] = React.useState<string[]>([])
+  const [showDropdown, setShowDropdown] = React.useState(false)
+  const [showSearchDropdown, setShowSearchDropdown] = React.useState(false)
+  const [searchValue, setSearchValue] = React.useState<string[]>([])
+
+  React.useEffect(() => {
+    const getFleets = transports.flatMap(({ metadata }) => metadata.fleet)
+    const fleetNames: string[] = getFleets
+      .filter((value, index) => getFleets.indexOf(value) === index)
+      .filter((name) => name !== '')
```

Comment:
This _could_ be:
```suggestion
      .filter((name) => Boolean(name))
```
since empty string is falsy